worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/WorldGuardEntityListener.java

```diff
@@ -653,35 +657,46 @@ public void onCreatureSpawn(CreatureSpawnEvent event) {
     public void onCreatePortal(PortalCreateEvent event) {
         WorldConfiguration wcfg = getWorldConfig(event.getWorld());
 
-        if (wcfg.regionNetherPortalProtection
+        if (wcfg.useRegions && wcfg.regionNetherPortalProtection
                 && event.getReason() == PortalCreateEvent.CreateReason.NETHER_PAIR
                 && !event.getBlocks().isEmpty()) {
             final com.sk89q.worldedit.world.World world = BukkitAdapter.adapt(event.getWorld());
-            final RegionManager regionManager = WorldGuard.getInstance().getPlatform().getRegionContainer()
-                    .get(world);
-            if (regionManager == null) return;
-            LocalPlayer associable = null;
-            if (event.getEntity() instanceof Player player) {
-                associable = getPlugin().wrapPlayer(player);
-                if (WorldGuard.getInstance().getPlatform().getSessionManager().hasBypass(associable, world)) {
+            final Cause cause = Cause.create(event.getEntity());
+            LocalPlayer localPlayer = null;
+            if (cause.getRootCause() instanceof Player player) {
+                if (wcfg.fakePlayerBuildOverride && InteropUtils.isFakePlayer(player)) {
+                    return;
+                }
+                localPlayer = getPlugin().wrapPlayer(player);
+                if (WorldGuard.getInstance().getPlatform().getSessionManager().hasBypass(localPlayer, world)) {
                     return;
                 }
             }
-            BlockVector3 min = null;
-            BlockVector3 max = null;
-            for (BlockState block : event.getBlocks()) {
-                BlockVector3 loc = BlockVector3.at(block.getX(), block.getY(), block.getZ());
-                min = min == null ? loc : loc.getMinimum(min);
-                max = max == null ? loc : loc.getMaximum(max);
-            }
-            ProtectedCuboidRegion target = new ProtectedCuboidRegion("__portal_check", true, min, max);
-            final ApplicableRegionSet regions = regionManager.getApplicableRegions(target);
-            if (!regions.testState(associable, Flags.BUILD, Flags.BLOCK_PLACE)) {
-                if (associable != null) {
+            final RegionManager regionManager = WorldGuard.getInstance().getPlatform().getRegionContainer()
+                    .get(world);
+            ApplicableRegionSet regions;
+            if (regionManager == null) {
+                regions = FailedLoadRegionSet.getInstance();
```

Comment:
isn't regionmanager also null if use-regions is false? in which case this should be the permissive set. really feels like we should delegate this to regionquery since it has this logic already.

Reply:
I've handled that case above already.

Reply:
> really feels like we should delegate this to regionquery since it has this logic already.

I'm not exactly sure what you mean by "delegate this" here. RQ isn't used here because we don't query regions at a specific location but for the whole portal region. Should I add a method to RQ for that too?